catroid/src/org/catrobat/catroid/content/Project.java

```diff
@@ -130,6 +136,14 @@ public synchronized boolean removeSprite(Sprite sprite) {
 		return spriteList;
 	}
 
+	public List<Sprite> getSpriteListWithClones() {
+		if (StageActivity.stageListener != null) {
+			return StageActivity.stageListener.getSpritesFromStage();
+		} else { // e.g. for ActionTests there is no Stage, only use sprites from Project
```

Comment:
correct. This method should never be called if there is no stage. If somebody calls is anyway (for some reason or by mistake) he gets the standard sprite-list, since there can't be any clones without a stage.